Notification dev bug on Android

We missed the condition where Notifcation Channels are only available from Android O (API 26) even though the minimum one is APi 23

```
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: de.tutao.tutanota.test:pushprocess, PID: 25263
    java.lang.NoClassDefFoundError: Failed resolution of: Landroid/app/NotificationChannel;
        at de.tutao.tutanota.push.LocalNotificationsFacade.createNotificationChannels(LocalNotificationsFacade.kt:226)
        at de.tutao.tutanota.push.PushNotificationService.onCreate(PushNotificationService.kt:107)
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:3017)
        at android.app.ActivityThread.access$1900(ActivityThread.java:150)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1495)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:168)
        at android.app.ActivityThread.main(ActivityThread.java:5885)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:797)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:687)
     Caused by: java.lang.ClassNotFoundException: Didn't find class "android.app.NotificationChannel" on path: DexPathList[[zip file "/data/app/de.tutao.tutanota.test-1/base.apk"],nativeLibraryDirectories=[/data/app/de.tutao.tutanota.test-1/lib/arm64, /data/app/de.tutao.tutanota.test-1/base.apk!/lib/arm64-v8a, /vendor/lib64, /system/lib64]]
        at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
        at de.tutao.tutanota.push.LocalNotificationsFacade.createNotificationChannels(LocalNotificationsFacade.kt:226) 
        at de.tutao.tutanota.push.PushNotificationService.onCreate(PushNotificationService.kt:107) 
        at android.app.ActivityThread.handleCreateService(ActivityThread.java:3017) 
        at android.app.ActivityThread.access$1900(ActivityThread.java:150) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1495) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:168) 
        at android.app.ActivityThread.main(ActivityThread.java:5885) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:797) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:687) 
    	Suppressed: java.lang.ClassNotFoundException: android.app.NotificationChannel
        at java.lang.Class.classForName(Native Method)
        at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
        at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
        		... 12 more
     Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
```

--- a/app-android/app/src/main/java/de/tutao/tutanota/AndroidFileFacade.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/AndroidFileFacade.kt
@@ -1,7 +1,6 @@
 package de.tutao.tutanota
 
 import android.Manifest
-import android.annotation.TargetApi
 import android.app.Activity
 import android.content.ContentValues
 import android.content.Context
@@ -12,6 +11,7 @@ import android.os.Environment
 import android.provider.MediaStore
 import android.util.Log
 import android.webkit.MimeTypeMap
+import androidx.annotation.RequiresApi
 import androidx.core.content.FileProvider
 import androidx.core.net.toUri
 import de.tutao.tutanota.ipc.*
@@ -142,7 +142,7 @@ class AndroidFileFacade(
 		}
 	}
 
-	@TargetApi(Build.VERSION_CODES.Q)
+	@RequiresApi(Build.VERSION_CODES.Q)
 	private suspend fun addFileToDownloadsMediaStore(fileUriString: String): String {
 		val contentResolver = activity.contentResolver
 		val fileUri = fileUriString.toUri()

--- a/app-android/app/src/main/java/de/tutao/tutanota/push/LocalNotificationsFacade.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/push/LocalNotificationsFacade.kt
@@ -11,6 +11,7 @@ import android.media.RingtoneManager
 import android.os.Build
 import android.util.Log
 import androidx.annotation.ColorInt
+import androidx.annotation.RequiresApi
 import androidx.annotation.StringRes
 import androidx.core.app.NotificationCompat
 import androidx.core.app.NotificationManagerCompat
@@ -221,7 +222,7 @@ class LocalNotificationsFacade(private val context: Context) {
 		notificationManager.notify(1000, notification)
 	}
 
-	@TargetApi(Build.VERSION_CODES.O)
+	@RequiresApi(Build.VERSION_CODES.O)
 	fun createNotificationChannels() {
 		val mailNotificationChannel = NotificationChannel(
 				EMAIL_NOTIFICATION_CHANNEL_ID,
@@ -252,7 +253,7 @@ class LocalNotificationsFacade(private val context: Context) {
 		notificationManager.createNotificationChannel(downloadNotificationsChannel)
 	}
 
-	@TargetApi(Build.VERSION_CODES.O)
+	@RequiresApi(Build.VERSION_CODES.O)
 	private fun NotificationChannel.default(): NotificationChannel {
 		val ringtoneUri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
 		val att = AudioAttributes.Builder()

--- a/app-android/app/src/main/java/de/tutao/tutanota/push/PushNotificationService.kt
+++ b/app-android/app/src/main/java/de/tutao/tutanota/push/PushNotificationService.kt
@@ -104,7 +104,9 @@ class PushNotificationService : LifecycleJobService() {
 			}
 		}
 
-		localNotificationsFacade.createNotificationChannels()
+		if (atLeastOreo()) {
+			localNotificationsFacade.createNotificationChannels()
+		}
 	}
 
 	private fun removeForegroundNotification() {
@@ -192,8 +194,7 @@ class PushNotificationService : LifecycleJobService() {
 		private const val SENDER_EXTRA = "sender"
 		const val ATTEMPT_FOREGROUND_EXTRA = "attemptForeground"
 
-		fun startIntent
-				(context: Context?, sender: String?, attemptForeground: Boolean): Intent {
+		fun startIntent(context: Context?, sender: String?, attemptForeground: Boolean): Intent {
 			val intent = Intent(context, PushNotificationService::class.java)
 			intent.putExtra(SENDER_EXTRA, sender)
 			intent.putExtra(ATTEMPT_FOREGROUND_EXTRA, attemptForeground)

